templates/types/streaming/nextjs/app/components/ui/chat/chat-message/chat-agent-events.tsx

```diff
@@ -20,10 +21,18 @@ const AgentIcons: Record<string, LucideIcon> = {
   publisher: icons.BookCheck,
 };
 
+type ProgressData = {
+  progress_id: string;
+  total_steps: number;
+  current_step: number;
+  step_msg: string;
+};
+
 type MergedEvent = {
   agent: string;
   texts: string[];
   icon: LucideIcon;
+  progress?: ProgressData;
```

Comment:
isn't that the right type?
```suggestion
  icon: LucideIcon;
  data: Array<string | ProgressData>;
```

Reply:
Yes, the type is correct. I just add a new progress field if it has progress bar.

Reply:
My idea was it's still the normal text event and if the event has progress properties then we will render the progress bar, i think easier to handle with that way and don't need to check what is the type of the data, additionally, we still have the whole text events of the agent after all and user can review what are the events of the progress.

Reply:
So progress is always implicitly just for the last message?

Reply:
I think you're talking about this one: https://github.com/run-llama/create-llama/pull/403#discussion_r1822043308
If it's then it's updated.

Reply:
@leehuwuj it looks way better now, we can close this